### PR TITLE
Let user know about dead instances in deferred calls (reverted)

### LINKED ISSUE #70910: Error spam (and/or crash) in release template / exported builds: `"slot >= slot_max" is true. Returning: nullptr`
### Godot version

4.0 beta 10

### System information

Windows 10 

### Issue description

In my rss reader on windows, when the user switches between feed lists - that is, when a VBoxContainer has all its children removed and a new set re-added, the application crashes with:

USER ERROR: Condition "slot >= slot_max" is true. Returning: nullptr
   at: get_instance (./core/object/object.h:974)

This only happens on the windows version and only when that version is built in release mode. In an earlier beta (7 or 10 I am not sure) it was happening occasionally on the linux build too, but that no longer appears to be an issue. I was not able to find any information or other bug reports about this error.

### Steps to reproduce

I have no simple minimal reproduction available, but it is fairly constant in my current project. You can find the source here:
https://gitlab.com/gametheatre/poriferareader/Porifera

1. Build a release windows version of the project
2. Run the exported executable
3. After running the project, subscribe to two or more RSS feeds.
4. If it still has not crashed, switch back and forth between these feeds.

Because this appears to be an issue with the build system it is likely that it will be a major hassle to debug. Let me know if there is any help I can offer.

### Minimal reproduction project

As mentioned above it is consistently reproducible with this (non minimal) project: https://gitlab.com/gametheatre/poriferareader/Porifera

As a temporary workaround, only creating debug builds for windows solves this.

### LINKED ISSUE #75802: Exported Version of Game Crashes at Random When Switching Scenes/Levels (slot >= slotmax)
### Godot version

v4.0.2.stable.official.7a0977ce2

### System information

Windows 10

### Issue description

I started a brand new project last week and I was testing an exported version of the game just to test my scene switcher and to make sure that everything was working as intended (in a previous Godot 4 project I made the exported game would also crash randomly when switching scenes). After exporting the game and running it, the scene switcher worked great for the first few attempts, and then crashed on about the 4th scene switch. I spent hours rewriting and reformatting my code and nothing seemed to work so I deleted the project and started from scratch, rebuilding everything I had made from the ground up. After running the new game in the editor, I then exported it and ran it just to be 100% sure that everything was working. I switched levels about 10 times and it worked flawlessly. I closed the .exe and re-ran it just to make sure that it wasn't a fluke and on the 3rd level change it crashed. I spent more time trying to fix it and nothing worked so I went onto the Godot discord for help and after a few hours of discussion, I finally found the culprit: 
slotmaxerror

The game runs perfectly fine in a debug export version of the game and in Godot itself, it's just the normal .exe export that's causing issues, which isn't good since that's obviously the version used for distributing the game publicly.

When I looked up the error in the Godot GitHub it became apparent that one of my resource IDs was corrupted, which shouldn't be possible considering this project is obviously a completely different project file than the first one. This is super frustrating and is clearly beyond my ability to fix and I can't continue work on my game until the issue is resolved.

### Steps to reproduce

1.) Export a normal version of the game (NOT a debug export) and run it.

2.) The game starts in a top-down hub world where you use the arrow keys to move. Move towards
 the flag and when you get close to it a "play" button will appear in the top left. Press it to switch scenes from the hub world to level 1.

3.) Once you're in level 1, run towards the right until you see the flag at the end of the level. Run into it to change scenes back to the hub world. (use space to jump and left and right arrow keys to move)

4.) Repeat steps 2 and 3 until you crash. Switch between scenes/levels 10 times. The game should crash before you reach 10 the first time you launch it. If you switch levels/scenes 10 times and nothing happens, close the .exe and re-launch it and try again and it should crash on the 2nd-4th scene switch. On very rare occasions the game will crash when you press the play button in the hub world.

### Minimal reproduction project

[slotmaxerror.zip](https://github.com/godotengine/godot/files/11182054/slotmaxerror.zip)

### LINKED ISSUE #78788: VideoStreamPlayer causes random crashes on a Windows export without "Export with Debug"
### Godot version

4.1.beta3, 4.0.3.stable

### System information

Godot v4.1.beta3 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 Ti (NVIDIA; 31.0.15.3598) - Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz (6 Threads)

### Issue description

The game crashes on a Windows export without "Export with Debug" when I use VideoStreamPlayer node as a background in my scene.

I have created 2 simple scenes with VideoStreamPlayer and a button in each to switch between them. When I run the game in the editor, it works fine, when I export it with debug it works fine. As soon as I untick "Export with Debug" the game crashes randomly, sometimes after a few button clicks(changing scenes), sometimes on the first click... The console gives no errors whatsoever...

Tried to do it in both: Godot 4.0.3 and 4.1 beta3, the result is the same. 

I have also recreated this in Godot 3.5.2, the exported game without "Export with Debug" works fine, without crashes at all.

I have spent 4-5 hours yesterday searching internet and trying to fix it, but I couldn't. Tried different ffmpeg builds to convert the video, tried different video framerates, with audio and without audio... Looks like it's a Godot 4 issue.

### Steps to reproduce

1 - create a scene, add VideoStreamPlayer node, assign your video to it.
2 - create another scene with a VideoStreamPlayer node and assign your video to it.
3 - create buttons on each scene to switch between them.
4 - export your project for windows without "Export with Debug"
5 - play the game, after switching the scenes a couple of time, the game will crash.

### Minimal reproduction project

[Crash_Test_3.zip](https://github.com/godotengine/godot/files/11893147/Crash_Test_3.zip)

### PR DESCRIPTION
**Pull request by RandomShaper**
So users have a hint about why their games print weird error messages or crash on release.

Fixes #75802.
Maybe fixes #78788.
Maybe fixes #70910.

### CODE REVIEW

File: core/object/message_queue.cpp [hunk 1]
```diff
@@ -34,6 +34,9 @@
 #include "core/core_string_names.h"
 #include "core/object/class_db.h"
 #include "core/object/script_language.h"
+#ifdef DEBUG_ENABLED
+#include "core/config/engine.h"
+#endif
```
Comment (akien-mga) [inline core/object/message_queue.cpp:39]: Style wise, add an empty line of separation between the regular list of include and conditionally defined ones:
```suggestion

#ifdef DEBUG_ENABLED
#include "core/config/engine.h"
#endif
```

**Comment by RandomShaper** [comment]
This one would be good for 4.2 dev builds.

**Review by clayjohn — APPROVED** [review]
This makes sense and seems very beneficial

**Comment by YuriSizov** [comment]
So we also get a lot of "ERROR: Trying to execute a deferred call/notification/set on a previously freed instance. Consider using queue_free() instead of free()." from running tests (you can see the CI run for this very PR). How should we avoid that?

**Comment by YuriSizov** [comment]
We also get this error in an empty Project Manager on startup. And I get it from simply opening a project in the editor too.

WindowsTerminal_2023-07-28_14-20-18

**Comment by reduz** [comment]
I am sorry, but this also has to be either reverted or changed. It is far too common for users to run into a situation where this will throw an error. This will frustrate more users than it will help because the errors are unfixable most of the time.

I did not add error checking for a reason here, which is that its *extremely* difficult to prevent these situations in a lot of scenarios.

**Comment by reduz** [comment]
I propose the following courses of action:

1) Revert this PR
2) Because you can't add new arguments here, you will need to add a new call type, either "call_validated" or "call_unvalidated". 
3) If the later, then all the usages of this calls in the engine will need to be renamed to "call_unvalidated" and the error printed will let the user know that this if this is intended, they should use call_unvalidated. If the former, do nothing.
4) Redo another PR doing this properly.

**Comment by reduz** [comment]
Remember that also we have method.call_deferred() in callable too, so whathever is done, a new version of this needs to be added there too. Also on the multithreading code. Adding validated/unvalidated versions sounds like a lot of work.

**Comment by YuriSizov** [comment]
This is reverted by https://github.com/godotengine/godot/pull/80081.

**Comment by AThousandShips** [comment]
Can't remember exactly when the "out of space" error for godot-cpp started popping up, but it seems to be at some point *after* this was merged, could the space issue be due to the log being filled with these messages from running the unit tests? Would be good to see if rebasing currently struggling PRs after merging #80081

**Comment by AThousandShips** [comment]
NVM realized #80081 still fails with that, I'm too tired to think clearly atm

**Comment by YuriSizov** [comment]
@AThousandShips What are you referring to?

**Comment by AThousandShips** [comment]
Went down a wrong rabbit hole with wondering if the log was flooded by this causing the space issues for the godot-cpp build, accidentally deleted instead of editing and only this comment got left

**Comment by AThousandShips** [comment]
Worth seeing if it reduces the frequency, but realized both that this occured before this got merged and also affects the revert PR so not directly causal

**Comment by jordanlis** [comment]
Has it been merged to 4.2dev2 ? I had no issues with 4.2dev1, now my code is buggy with this when I use set_deffered, I don't understand how to solve it. 

Here's a conversation that I found on reddit for information https://www.reddit.com/r/godot/comments/15jhppb/please_help_constant_issues_with_queuefree/?sort=new

**Comment by AThousandShips** [comment]
It has been reverted, https://github.com/godotengine/godot/pull/80081, so it will be fixed in dev3

**Pull request merged**